### PR TITLE
Add ability to pass shared library name to Player

### LINKED ISSUE #65: Possible strange behaviour with self interactions?
Looking at `k42r` (which is marked as not Stochastic) does not play against itself in a symmetric way:

```python
>>> players = (axlf.Player("k42r"), axlf.Player("k42r"))
>>> match = axl.Match(players, turns=200)
>>> results = match.play()
>>> results[25:30]
[(C, C), (C, D), (D, D), (D, D), (D, D)]
```

So on turn 27 (`(C, D)`) the second `k42r` defects: which obviously it should not because the histories up until then are the same (have checked: this is not a stochastic effect).

I think this requires carefully walking through the strategy but points towards perhaps a problem with self interactions?

### PR DESCRIPTION
**Pull request by drvinceknight**
I have also added a warning when a self interaction is being played.
Not that the warning only happens once (as indicated by the test).

I have also removed the tournament in our README.

Closes #65

### CODE REVIEW

**Comment by drvinceknight** [comment]
FYI, this is what the warning looks like in a notebook:

screenshot from 2017-10-19 14-12-32

File: src/axelrod_fortran/player.py [hunk 1]
```diff
@@ -72,6 +75,18 @@ def original_strategy(
         return self.original_function(*[byref(arg) for arg in args])
 
     def strategy(self, opponent):
+        if type(opponent) is Player\
```
Comment (marcharper) [inline src/axelrod_fortran/player.py]: space after player please, indentation looks like it's off by one character (but maybe not)

**Comment by marcharper** [comment]
This is fine to merge, though I wonder if there is a better solution, such as automatically generating a new clone every time `Player(name)` is invoked. I realize this might use a lot of memory, but I think it would not be hard to reuse shared library instances as needed for new clones with some kind of registry. Just a thought (not asking for this PR to be altered).

**Comment by drvinceknight** [comment]
When you get a chance @meatballs if you could merge and release I'll get the data collection going again :) :+1:

**Pull request merged**